tfhe/src/integer/server_key/radix_parallel/add.rs

```diff
@@ -232,12 +232,94 @@ impl ServerKey {
         };
 
         if self.is_eligible_for_parallel_single_carry_propagation(lhs) {
-            self.unchecked_add_assign_parallelized_low_latency(lhs, rhs);
+            let _carry = self.unchecked_add_assign_parallelized_low_latency(lhs, rhs);
         } else {
             self.unchecked_add_assign(lhs, rhs);
             self.full_propagate_parallelized(lhs);
         }
     }
+    /// Computes the addition of two unsigned ciphertexts and returns the overflow flag
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use tfhe::integer::gen_keys_radix;
+    /// use tfhe::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+    ///
+    /// // Generate the client key and the server key:
+    /// let num_blocks = 4;
+    /// let (cks, sks) = gen_keys_radix(PARAM_MESSAGE_2_CARRY_2_KS_PBS, num_blocks);
+    ///
+    /// let msg1 = u8::MAX;
+    /// let msg2 = 1;
+    ///
+    /// let ct1 = cks.encrypt(msg1);
+    /// let ct2 = cks.encrypt(msg2);
+    ///
+    /// let (ct_res, overflowed) = sks.unsigned_overflowing_add_parallelized(&ct1, &ct2);
+    ///
+    /// // Decrypt:
+    /// let dec_result: u8 = cks.decrypt(&ct_res);
+    /// let dec_overflowed = cks.decrypt_one_block(&overflowed);
+    /// let (expected_result, expected_overflow) = msg1.overflowing_add(msg2);
+    /// assert_eq!(dec_result, expected_result);
+    /// assert_eq!(dec_overflowed, u64::from(expected_overflow));
+    /// ```
+    pub fn unsigned_overflowing_add_parallelized(
+        &self,
+        ct_left: &RadixCiphertext,
+        ct_right: &RadixCiphertext,
+    ) -> (RadixCiphertext, Ciphertext) {
```

Comment:
should the overflow flag be a boolean block or a plain ciphertext ?

I know there is likely the question of a potential PBS in there

I guess the question is: does the carry block contain a 1 or a 0 ? if so then we can likely wrap it in a boolean block

Reply:
I planned on making it a BooleanBlock after the merge (this branch is not up to date with the merge of booleanblock)
and also the 2 overflowing_sub do not return boolean block yet

Reply:
ok !